gammapy/spectrum/observation.py

```diff
@@ -518,22 +518,35 @@ def copy(self):
         """A deep copy."""
         return copy.deepcopy(self)
 
-    def to_spectrum_dataset(self):
+    def to_spectrum_dataset(self, model=None):
```

Comment:
As I've written below it's just meant to be a preliminary solution to help with the transition from `SpectrumObservation` to `SpectrumDatasetOnOff`. I just thought it was better to have this transition layer in one single place. But thinking about it again it's probably better to set the model afterwards, because that's what we have to do for the `SpectrumDatasetOnOff` anyway...I guess this is what you meant?

Reply:
I've looked into this again and the problem was that `SpectrumDataset` does not allow `model=None` on init. But it's probably the much better idea to change it there...